lib/serialization.ml

```diff
@@ -1,15 +1,18 @@
 module List = Core.List
 module String = Core.String
+module Option = Core.Option
 
 let show pub =
- List.reduce_exn pub ~f:Utils.concat_hashes
+  pub
+  |> List.map ~f:Utils.bytes_to_string
+  |> List.reduce_exn ~f:Utils.concat_hashes
+
 
 let load text =
   let list = String.split text ~on:'-' in
+  let open Option in
   Utils.validate_key list
+  >>= fun list -> some @@ List.map ~f:Utils.bytes_of_string list
 
-let address pub =
-  pub
-  |> show
-  |> Hash.digest
-  |> Utils.to_hex
+
+let digest pub = pub |> show |> Hash.digest
```

Comment:
I should rethink the serialization of public key and signature (text signature + verification key). By using Base64 instead Hexcodes to reduce the signature size, I will ending up using only newlines as separators/markers, so no `"-"` neither `"@"` would help me - I need the core library layer accepting a list of bytes/string instead. **The responsibility to serialize and parse should be deferred to the topmost layer possible to not pollute my domain logic neither to confuse/mix things potentially introducing bugs/security loopholes.**

It will possibly break some of my tests, so they will need some refactoring to be paired too.